nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AbstractDatabaseFetchProcessor.java

```diff
@@ -0,0 +1,361 @@
+
+            case TIME:
+                Date rawColTimeValue = resultSet.getDate(columnIndex);
+                java.sql.Time colTimeValue = new java.sql.Time(rawColTimeValue.getTime());
+                java.sql.Time maxTimeValue = null;
+                if (maxValueString != null) {
+                    maxTimeValue = java.sql.Time.valueOf(maxValueString);
+                }
+                if (maxTimeValue == null || colTimeValue.after(maxTimeValue)) {
+                    return colTimeValue.toString();
+                }
+                break;
+
+            case TIMESTAMP:
+                // Oracle timestamp queries must use literals in java.sql.Date format
+                if ("Oracle".equals(preProcessStrategy)) {
+                    Date rawColOracleTimestampValue = resultSet.getDate(columnIndex);
+                    java.sql.Date oracleTimestampValue = new java.sql.Date(rawColOracleTimestampValue.getTime());
+                    java.sql.Date maxOracleTimestampValue = null;
+                    if (maxValueString != null) {
+                        maxOracleTimestampValue = java.sql.Date.valueOf(maxValueString);
+                    }
+                    if (maxOracleTimestampValue == null || oracleTimestampValue.after(maxOracleTimestampValue)) {
+                        return oracleTimestampValue.toString();
+                    }
+                } else {
+                    Timestamp rawColTimestampValue = resultSet.getTimestamp(columnIndex);
+                    java.sql.Timestamp colTimestampValue = new java.sql.Timestamp(rawColTimestampValue.getTime());
+                    java.sql.Timestamp maxTimestampValue = null;
+                    if (maxValueString != null) {
+                        maxTimestampValue = java.sql.Timestamp.valueOf(maxValueString);
+                    }
+                    if (maxTimestampValue == null || colTimestampValue.after(maxTimestampValue)) {
+                        return colTimestampValue.toString();
+                    }
+                }
+                break;
+
+            case BIT:
+            case BOOLEAN:
+            case BINARY:
+            case VARBINARY:
+            case LONGVARBINARY:
+            case ARRAY:
+            case BLOB:
+            case CLOB:
+            default:
+                throw new IOException("Type for column " + columnIndex + " is not valid for maintaining maximum value");
+        }
+        return null;
+    }
+
+    /**
+     * Returns a SQL literal for the given value based on its type. For example, values of character type need to be enclosed
+     * in single quotes, whereas values of numeric type should not be.
+     *
+     * @param type  The JDBC type for the desired literal
+     * @param value The value to be converted to a SQL literal
+     * @return A String representing the given value as a literal of the given type
+     */
+    protected static String getLiteralByType(int type, String value, String preProcessStrategy) {
```

Comment:
The last parameter shouldn't be "preProcessStrategy" since that property was removed in favor of database type.